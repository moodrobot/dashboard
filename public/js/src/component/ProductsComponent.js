import React, { PureComponent } from "react";
import PropTypes from "prop-types";

class ProductsComponent extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      style: "category",
      data: this.props.data,
      loader_display: "",
      view_more_display: "d-none",
      content_display: "d-none",
      no_content_display: "d-none",
      max: 10,
      display_num: 1
    };

    this.showMoreItems = this.showMoreItems.bind(this);
  }

  showMoreItems() {
    var count = this.state.display_num;
    count++;

    //set new display amount and re-render
    this.setState({ display_num: count });
  }

  static getDerivedStateFromProps(nextProps, state) {
    //HIDES THE PRELOADER

    //view more display
    let vm = "";
    let cc = "d-block";
    let no_cont = "d-none";

    if (nextProps.init) {
      
      if (nextProps.data.length < state.max) {
        //document.getElementById("view_more_prod_btn").style.display = "none";
        vm = "d-none";
      }

      if (nextProps.data.length === 0) {
        //document.getElementById("no_prod").style.display = "block";
        no_cont = "d-block";
        cc = "d-none";
      }

      if (state.display_num * state.max >= nextProps.data.length && nextProps.data.length > 0) {
        vm = "d-none";
      } else if (nextProps.data.length >= nextProps.max) {
        vm = "";
      }

      return {
        data: nextProps.data,
        loader_display: "d-none",
        content_display: cc,
        view_more_display: vm,
        no_content_display: no_cont
      };
    } else {
      return null;
    }
  }

  /*componentDidUpdate() {
    let vm;

    if (this.state.display_num * this.state.max >= this.state.data.length && this.state.data.length > 0) {
      vm = "d-none";
    } else if (this.state.data.length >= this.state.max) {
      vm = "";
    }

    //MAKE SURE THERE IS A CHANGE OR WILL INFINITE LOOP
    if (vm != this.state.view_more_display) {
      this.setState({ view_more_display: vm });
    }
  }*/

  createOptions() {
    const T = this.props.translate;
    let currentLanguage = this.props.currentLanguage;

    const listItems = this.state.data.slice(0, this.state.max * this.state.display_num).map(function(item, index) {
      let licenseLink = "";
      if (item.moreinfourl) {
        licenseLink = (
          <p className="top-pad">
            <a href={item.moreinfourl}>{T["manage-license"]}</a>
          </p>
        );
      }

      let forumLink = "";
      if (item.forumurl) {
        forumLink = (
          <p className="top-pad">
            <a href={item.forumurl}>{T["forum"]}</a>
          </p>
        );
      }

      let downloadLink = "";
      if (item.download) {
        downloadLink = (
          <a href={item.download} className="btn btn-orange">
            {T["download-button"]}
          </a>
        );
      }

      return (
        <li className="product" key={index} value={item.value}>
          <div className="row">
            <div className="col-md-4 col-sm-12 text-center">
              <img className="float-image-center" src={item.imageurl} />
            </div>
            <div className="col">
              <div className="row no-gutters">
                <div className="col my-auto">
                  <span className="title text-md-nowrap">{item.name}</span>
                </div>
                <div className="col-md-auto">{downloadLink}</div>
              </div>
              <div className="row align-top">
                <div className="col-sm-7 col-lg-5 category">{T["serial"]}</div>
                <div className="col-sm-5 content">{item.serialnumber}</div>
              </div>
              <div className="row align-top">
                <div className="col-sm-7 col-lg-5 category">{T["license-type"]}</div>
                <div className="col-sm-5 content">{item.licensetype}</div>
              </div>
              <div className="row align-top">
                <div className="col-sm-7 col-lg-5 category">{T["purchase-date"]}</div>
                <div className="col-sm-5 content">
                  {new Date(item.registerdateepoch * 1000).toLocaleDateString(currentLanguage)}
                </div>
              </div>
              <div className="row align-top">
                <div className="col-sm-7 col-lg-5 category">{T["expiry-date"]}</div>
                <div className="col-sm-5 content">
                  {new Date(item.supportexpirationdateepoch * 1000).toLocaleDateString(currentLanguage)}
                </div>
              </div>
              {licenseLink}
              {forumLink}
            </div>
          </div>
        </li>
      );
    });
    return listItems;
  }

  render() {
    //console.log("RENDERING PRODUCTS COMPONENT");

    return (
      <div className="container-fluid">
        <div className="loader row">
          <div id="no_prod" className={this.state.no_content_display}>
            No products yet
          </div>
          <div id="loader_prod" className={"col-12 " + this.state.loader_display}>
            <img src="images/loader.gif" />
          </div>
        </div>
        <div id="product-content" className={this.state.content_display}>
          <div className="container-fluid">
            <ul className="list-unstyled">{this.createOptions()}</ul>
          </div>
          <div className="row text-center">
            <div className="col text-center">
              <button
                id="view_more_prod_btn"
                onClick={this.showMoreItems}
                type="button"
                className={"btn btn-secondary " + this.state.view_more_display}
              >
                {this.props.view_more}
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

ProductsComponent.propTypes = {
  translate: PropTypes.object,
  data: PropTypes.array,
  currentLanguage: PropTypes.string,
  view_more: PropTypes.string,
  init: PropTypes.bool
};

export default ProductsComponent;
