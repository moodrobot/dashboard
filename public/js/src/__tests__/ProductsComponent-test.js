import React from "react";
import { shallow, mount } from "enzyme";
import sinon from "sinon";
import ProductsComponent from "../component/ProductsComponent";
import productsData from "./__mockData__/productsData.json";
import translateData from "../../../../locales/en-US.json";
import i18n from "../i18n-for-testing"; // initialized i18next instance for testing

describe("<ProductsComponent />", () => {
  const wrapper = shallow(
    <ProductsComponent
      data={productsData}
      currentLanguage={"en-US"}
      translate={i18n.t("products")}
      view_more = {i18n.t("view-more-button")}
      init={true}
    />
  );


  it("expect to match snapshot", () => {
    /**EXPECT NO CHANGE IN THE SNAPSHOT */
    expect(wrapper).toMatchSnapshot();
  });

  it("expect display number to incerease and show 20 total list items", () => {
    /**CLICK THE VIEW MORE BUTTON */
    wrapper.find("button").first().simulate("click");

    wrapper.update();

    /**EXPECT THE DISPLAY NUM TO INCREASE BY 1 */
    expect(wrapper.state().display_num).toBe(2);

    //EXPECT THE LIST TO SHOW AS MANY ITEMS AS THE MAX MULTIPLIED BY THE DISPLAY NUMBER
    expect(wrapper.find("ul").find("li").length).toBe(20);//.debug());

  });
});
