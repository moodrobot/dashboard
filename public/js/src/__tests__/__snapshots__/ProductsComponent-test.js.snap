// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsComponent /> expect to match snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="loader row"
  >
    <div
      className="d-none"
      id="no_prod"
    >
      No products yet
    </div>
    <div
      className="col-12 d-none"
      id="loader_prod"
    >
      <img
        src="images/loader.gif"
      />
    </div>
  </div>
  <div
    className="d-block"
    id="product-content"
  >
    <div
      className="container-fluid"
    >
      <ul
        className="list-unstyled"
      >
        <li
          className="product"
          key="0"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4 col-sm-12 text-center"
            >
              <img
                className="float-image-center"
                src="https://registrationcenter.intel.com/includes/images/boxshots/BS_GO.png"
              />
            </div>
            <div
              className="col"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col my-auto"
                >
                  <span
                    className="title text-md-nowrap"
                  >
                    Intel® GO™ Automotive SDK
                  </span>
                </div>
                <div
                  className="col-md-auto"
                />
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Serial Number
                </div>
                <div
                  className="col-sm-5 content"
                >
                  B7HV-R7B874GG
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  License Type
                </div>
                <div
                  className="col-sm-5 content"
                >
                  Named-user
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Purchase Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  10/4/2017
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Expiration Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  6/30/2018
                </div>
              </div>
              <p
                className="top-pad"
              >
                <a
                  href="https://registrationcenter.intel.com/en/products/subscription/834"
                >
                  Manage License
                </a>
              </p>
              <p
                className="top-pad"
              >
                <a
                  href="https://software.intel.com/en-us/go-automotive-sdk/support"
                >
                  Forum
                </a>
              </p>
            </div>
          </div>
        </li>
        <li
          className="product"
          key="1"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4 col-sm-12 text-center"
            >
              <img
                className="float-image-center"
                src="https://registrationcenter.intel.com/includes/images/boxshots/BS_Marquee_system_studio.png"
              />
            </div>
            <div
              className="col"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col my-auto"
                >
                  <span
                    className="title text-md-nowrap"
                  >
                    Intel® System Studio Beta
                  </span>
                </div>
                <div
                  className="col-md-auto"
                />
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Serial Number
                </div>
                <div
                  className="col-sm-5 content"
                >
                  B7H4-XTDG54FF
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  License Type
                </div>
                <div
                  className="col-sm-5 content"
                >
                  Named-user
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Purchase Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  10/2/2017
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Expiration Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  1/6/2018
                </div>
              </div>
              <p
                className="top-pad"
              >
                <a
                  href="https://registrationcenter.intel.com/en/products/subscription/827"
                >
                  Manage License
                </a>
              </p>
              <p
                className="top-pad"
              >
                <a
                  href="https://software.intel.com/en-us/forums/intel-system-studio"
                >
                  Forum
                </a>
              </p>
            </div>
          </div>
        </li>
        <li
          className="product"
          key="2"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4 col-sm-12 text-center"
            >
              <img
                className="float-image-center"
                src="https://registrationcenter.intel.com/includes/images/boxshots/BS_Python.png"
              />
            </div>
            <div
              className="col"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col my-auto"
                >
                  <span
                    className="title text-md-nowrap"
                  >
                    Intel® Distribution for Python
                  </span>
                </div>
                <div
                  className="col-md-auto"
                />
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Serial Number
                </div>
                <div
                  className="col-sm-5 content"
                >
                  CTGC-JJPDBCTZ
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  License Type
                </div>
                <div
                  className="col-sm-5 content"
                >
                  No License
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Purchase Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  6/29/2017
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Expiration Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  6/29/2018
                </div>
              </div>
              <p
                className="top-pad"
              >
                <a
                  href="https://registrationcenter.intel.com/en/products/subscription/797"
                >
                  Manage License
                </a>
              </p>
              <p
                className="top-pad"
              >
                <a
                  href="https://software.intel.com/en-us/forums/intel-distribution-for-python"
                >
                  Forum
                </a>
              </p>
            </div>
          </div>
        </li>
        <li
          className="product"
          key="3"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4 col-sm-12 text-center"
            >
              <img
                className="float-image-center"
                src="https://registrationcenter.intel.com/includes/images/boxshots/BS_Python.png"
              />
            </div>
            <div
              className="col"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col my-auto"
                >
                  <span
                    className="title text-md-nowrap"
                  >
                    Intel® Distribution for Python
                  </span>
                </div>
                <div
                  className="col-md-auto"
                />
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Serial Number
                </div>
                <div
                  className="col-sm-5 content"
                >
                  CTGC-ZGS5HSTT
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  License Type
                </div>
                <div
                  className="col-sm-5 content"
                >
                  No License
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Purchase Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  6/8/2017
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Expiration Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  6/8/2018
                </div>
              </div>
              <p
                className="top-pad"
              >
                <a
                  href="https://registrationcenter.intel.com/en/products/subscription/797"
                >
                  Manage License
                </a>
              </p>
              <p
                className="top-pad"
              >
                <a
                  href="https://software.intel.com/en-us/forums/intel-distribution-for-python"
                >
                  Forum
                </a>
              </p>
            </div>
          </div>
        </li>
        <li
          className="product"
          key="4"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4 col-sm-12 text-center"
            >
              <img
                className="float-image-center"
                src="https://registrationcenter.intel.com/includes/images/boxshots/BS_Python.png"
              />
            </div>
            <div
              className="col"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col my-auto"
                >
                  <span
                    className="title text-md-nowrap"
                  >
                    Intel® Distribution for Python
                  </span>
                </div>
                <div
                  className="col-md-auto"
                />
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Serial Number
                </div>
                <div
                  className="col-sm-5 content"
                >
                  CTGC-PK25ZR6L
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  License Type
                </div>
                <div
                  className="col-sm-5 content"
                >
                  No License
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Purchase Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  6/8/2017
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Expiration Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  6/8/2018
                </div>
              </div>
              <p
                className="top-pad"
              >
                <a
                  href="https://registrationcenter.intel.com/en/products/subscription/797"
                >
                  Manage License
                </a>
              </p>
              <p
                className="top-pad"
              >
                <a
                  href="https://software.intel.com/en-us/forums/intel-distribution-for-python"
                >
                  Forum
                </a>
              </p>
            </div>
          </div>
        </li>
        <li
          className="product"
          key="5"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4 col-sm-12 text-center"
            >
              <img
                className="float-image-center"
                src="https://registrationcenter.intel.com/includes/images/boxshots/BS_XE_edition.png"
              />
            </div>
            <div
              className="col"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col my-auto"
                >
                  <span
                    className="title text-md-nowrap"
                  >
                    Intel® Parallel Studio XE Cluster Edition for Linux*
                  </span>
                </div>
                <div
                  className="col-md-auto"
                />
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Serial Number
                </div>
                <div
                  className="col-sm-5 content"
                >
                  C9TX-DZ97JCLM
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  License Type
                </div>
                <div
                  className="col-sm-5 content"
                >
                  Named-user
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Purchase Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  4/25/2017
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Expiration Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  4/25/2018
                </div>
              </div>
              <p
                className="top-pad"
              >
                <a
                  href="https://registrationcenter.intel.com/en/products/subscription/441"
                >
                  Manage License
                </a>
              </p>
              <p
                className="top-pad"
              >
                <a
                  href="https://software.intel.com/en-us/forums/intel-software-development-products"
                >
                  Forum
                </a>
              </p>
            </div>
          </div>
        </li>
        <li
          className="product"
          key="6"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4 col-sm-12 text-center"
            >
              <img
                className="float-image-center"
                src="https://registrationcenter.intel.com/includes/images/boxshots/BS_XE_edition.png"
              />
            </div>
            <div
              className="col"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col my-auto"
                >
                  <span
                    className="title text-md-nowrap"
                  >
                    Intel® Parallel Studio XE Beta 
                  </span>
                </div>
                <div
                  className="col-md-auto"
                />
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Serial Number
                </div>
                <div
                  className="col-sm-5 content"
                >
                  B2C3-X8JG2DP9
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  License Type
                </div>
                <div
                  className="col-sm-5 content"
                >
                  Named-user
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Purchase Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  4/21/2017
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Expiration Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  10/12/2017
                </div>
              </div>
              <p
                className="top-pad"
              >
                <a
                  href="https://registrationcenter.intel.com/en/products/subscription/690"
                >
                  Manage License
                </a>
              </p>
            </div>
          </div>
        </li>
        <li
          className="product"
          key="7"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4 col-sm-12 text-center"
            >
              <img
                className="float-image-center"
                src={null}
              />
            </div>
            <div
              className="col"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col my-auto"
                >
                  <span
                    className="title text-md-nowrap"
                  >
                    Intel® Capability Licensing Service
                  </span>
                </div>
                <div
                  className="col-md-auto"
                />
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Serial Number
                </div>
                <div
                  className="col-sm-5 content"
                >
                  C2N8-DVRVFXMN
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  License Type
                </div>
                <div
                  className="col-sm-5 content"
                >
                  No License
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Purchase Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  4/4/2017
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Expiration Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  4/4/2018
                </div>
              </div>
              <p
                className="top-pad"
              >
                <a
                  href="https://registrationcenter.intel.com/en/products/subscription/818"
                >
                  Manage License
                </a>
              </p>
              <p
                className="top-pad"
              >
                <a
                  href="https://software.intel.com/en-us/forums/intel-software-guard-extensions-intel-sgx"
                >
                  Forum
                </a>
              </p>
            </div>
          </div>
        </li>
        <li
          className="product"
          key="8"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4 col-sm-12 text-center"
            >
              <img
                className="float-image-center"
                src="https://registrationcenter.intel.com/includes/images/boxshots/BS_Marquee_Gears.png"
              />
            </div>
            <div
              className="col"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col my-auto"
                >
                  <span
                    className="title text-md-nowrap"
                  >
                    Deep Learning Training Tool Beta from Intel
                  </span>
                </div>
                <div
                  className="col-md-auto"
                />
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Serial Number
                </div>
                <div
                  className="col-sm-5 content"
                >
                  BBLF-XK4GVCNM
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  License Type
                </div>
                <div
                  className="col-sm-5 content"
                >
                  No License
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Purchase Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  3/31/2017
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Expiration Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  12/31/2017
                </div>
              </div>
              <p
                className="top-pad"
              >
                <a
                  href="https://registrationcenter.intel.com/en/products/subscription/800"
                >
                  Manage License
                </a>
              </p>
            </div>
          </div>
        </li>
        <li
          className="product"
          key="9"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4 col-sm-12 text-center"
            >
              <img
                className="float-image-center"
                src="https://registrationcenter.intel.com/includes/images/boxshots/BS_Marquee_Gears.png"
              />
            </div>
            <div
              className="col"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col my-auto"
                >
                  <span
                    className="title text-md-nowrap"
                  >
                    Intel® Performance Libraries
                  </span>
                </div>
                <div
                  className="col-md-auto"
                />
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Serial Number
                </div>
                <div
                  className="col-sm-5 content"
                >
                  33RM-RDF8T4JG
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  License Type
                </div>
                <div
                  className="col-sm-5 content"
                >
                  SystemLocked
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Purchase Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  3/1/2017
                </div>
              </div>
              <div
                className="row align-top"
              >
                <div
                  className="col-sm-7 col-lg-5 category"
                >
                  Expiration Date
                </div>
                <div
                  className="col-sm-5 content"
                >
                  3/1/2018
                </div>
              </div>
              <p
                className="top-pad"
              >
                <a
                  href="https://registrationcenter.intel.com/en/products/subscription/737"
                >
                  Manage License
                </a>
              </p>
              <p
                className="top-pad"
              >
                <a
                  href="https://software.intel.com/en-us/user-support-forums-for-intel-performance-libraries"
                >
                  Forum
                </a>
              </p>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="row text-center"
    >
      <div
        className="col text-center"
      >
        <button
          className="btn btn-secondary "
          id="view_more_prod_btn"
          onClick={[Function]}
          type="button"
        >
          View More
        </button>
      </div>
    </div>
  </div>
</div>
`;
